universal-application-tool-0.0.1/app/views/admin/programs/ManageProgramAdminsView.java

```diff
@@ -47,7 +48,10 @@ public ManageProgramAdminsView(AdminLayout layout) {
 
   /** Display a form with a list of inputs for adding and removing admins. */
   public Content render(
-      Http.Request request, ProgramDefinition program, ImmutableList<String> existingAdminEmails) {
+      Http.Request request,
+      ProgramDefinition program,
+      ImmutableList<String> existingAdminEmails,
+      String message) {
```

Comment:
suggestion (blocking): IMO the intention of calling render would be more clear from the call site if you overloaded the method, leaving the original call site (ProgramAdminManagementController.java at L63) unchanged and changed the signature for this version of the method to use `Optional<String> message`, making it more obvious that the message is not always expected to be present.

Reply:
Great point. Implemented the overloading -- let me know if theres a better way to do it.

Question: What is the point of having `Optional<String> message` in this version of the method? If it's an overload method, why does the signature have to be Optional?

Reply:
It makes it clear that `message` is optional and the behavior of the method will change based on whether it's there or not. That's not readily apparent with a plain `String` and a `if` in the method body to check if it's empty.

Reply:
That makes sense to me. I'm still unclear on why we'd do this for an overloaded function. My sense is having two versions of the function makes it clear that the String message is not always needed. I would think that the Optional<String> signature would be helpful if we only have one version of the function. I haven't worked with overloaded functions much, so I could be off course here. 

Here's a simple example of how I'm thinking about it...

function printMe(string a) {
    print(a)
}

function printMe(string a, string b) { /// why make string b optional?
   print(a + b);
}

Thanks for your explanations here!

Reply:
Also let me know if theres a better way to do it (i.e. is it better to have a base function for the shared logic that each render() function calls?)

Reply:
Ah, yeah, it's a little awkward with both overloading and the optional. Feel to pick one or the other. Sorry for the churn!

Reply:
Sounds good! And no worries, Thanks for the suggestions.